Compiler warnings in wayland code

```
wl_window.c: In function ‘xdgToplevelHandleConfigure’:
wl_window.c:299:14: error: variable ‘resizing’ set but not used [-Werror=unused-but-set-variable]
     GLFWbool resizing = GLFW_FALSE;
```

And the issue I tried to fix in #1143 is still not fixed

```
wl_init.c: In function ‘pointerHandleAxis’:
wl_init.c:167:5: error: ‘y’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
     _glfwInputScroll(window, x, y);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
wl_init.c:167:5: error: ‘x’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
```

Is initializing x and y to zero, really such a big deal? The asserts dont make the warning go away when compiled as:

```
gcc -MMD -Wno-missing-field-initializers -Wall -std=c99 -D_XOPEN_SOURCE=700 -Werror -g3 -Og -DDEBUG -fwrapv -fstack-protector-strong -pipe -fvisibility=hidden -fPIC -pthread -fpic -D_GLFW_WAYLAND -D_GLFW_BUILD_DLL ...
```

--- a/src/wl_window.c
+++ b/src/wl_window.c
@@ -296,7 +296,6 @@ static void xdgToplevelHandleConfigure(void* data,
     uint32_t* state;
     GLFWbool maximized = GLFW_FALSE;
     GLFWbool fullscreen = GLFW_FALSE;
-    GLFWbool resizing = GLFW_FALSE;
     GLFWbool activated = GLFW_FALSE;
 
     wl_array_for_each(state, states)
@@ -310,7 +309,6 @@ static void xdgToplevelHandleConfigure(void* data,
                 fullscreen = GLFW_TRUE;
                 break;
             case XDG_TOPLEVEL_STATE_RESIZING:
-                resizing = GLFW_TRUE;
                 break;
             case XDG_TOPLEVEL_STATE_ACTIVATED:
                 activated = GLFW_TRUE;

